### PR TITLE
Analytics provider docs

### PR DESCRIPTION
**Pull request by liviur**
## Description

Refactor of analytics providers docs - removing a lot of irrelevant stuff and adding brand new guides for iOS and android

@snooky23 please take a DEEP look at the android one - I feel it's not as deep as iOS.
Feel free to send the links off the branch if you need them Sunday / Monday and freely change the android by pushing commits.

## Checklist

* [ ] PR is scoped to one task
* [ ] Tests are included
* [ ] Documentation is included

* [ ] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
* specific apps / plugins to test :

### CODE REVIEW

File: book/analytics/client_side_api/client_side_api.md [hunk 1]
```diff
@@ -59,49 +62,53 @@ Some characteristics of users should be stored as user properties[c]. This is pa
...
+  // sending event parameter  
+  AnalyticsAgentUtil.logEvent("Article Read", articleParams);
+```
 
 ### iOS:
-
-NSDictionary \*analyticsDictionary = @{@”Author” : @”"John Q", @User Status": @"Registered"};  
-[APAnalyticsManager trackEvent:@"Article Read" withParameters:analyticsDictionary];
+```
+  NSDictionary \*analyticsDictionary = @{@”Author” : @”"John Q", @User Status": @"Registered"};
```
Comment (lomudi) [inline book/analytics/client_side_api/client_side_api.md]: is the `\` in the `NSDictionary \*analyticsDictionary` is necessary?

File: book/analytics/client_side_api/client_side_api.md [hunk 2]
```diff
@@ -59,49 +62,53 @@ Some characteristics of users should be stored as user properties[c]. This is pa
...
+  AnalyticsAgentUtil.logTimedEvent("Article Read", articleParams);
+
+  // ...  
+  // End the timed event, when the user navigates away from article  
+  AnalyticsAgentUtil.endTimedEvent("Article Read");
+```
 
 ### iOS:
+```
+  NSDictionary \*analyticsDictionary = @{@”Author” : @”"John Q", @User Status": @"Registered" };
```
Comment (lomudi) [inline book/analytics/client_side_api/client_side_api.md]: is the `\` in the `NSDictionary \*analyticsDictionary` is necessary?

File: book/analytics/client_side_api/client_side_api.md (hunk 1 above)
Comment (liviur) [reply book/analytics/client_side_api/client_side_api.md]: 👍

File: book/analytics/client_side_api/client_side_api.md (hunk 2 above)
Comment (liviur) [reply book/analytics/client_side_api/client_side_api.md]: 👍

File: book/analytics/client_side_api/client_side_api.md [hunk 3]
```diff
@@ -176,21 +185,21 @@ Additionally, some analytics providers (like comScore, AGOF, or Nielsen), create
 For example:  
 IOS:
 
-```bash
+```obj-c
   APAnalyticsManager trackScreenView:@"Splash - App Loading Screen"];
```
[the remark this answers is not in the document]
Comment (liviur) [reply book/analytics/client_side_api/client_side_api.md]: 🌡

**Comment by liviur** [comment]
@lomudi fixed the mentioned issues.
They were from previous commits which I re-organized.

File: book/analytics/analytics_utility/analytics_utility.md [hunk 4]
```diff
@@ -35,19 +65,7 @@ For example, when you click on the Side Menu, you’ll see:
...
-4. You should now be able to see the alerts:
-	![alert_ios_5](./alert_ios_5.png)
-
-
-## Viewing Logs in the Console
-
-#### Android
+### Viewing Logs in the Console
 
 If you do not have Android Studio, you’ll need to download it. You can do so [here](https://developer.android.com/studio/index.html).
```
Comment (snooky23) [inline book/analytics/analytics_utility/analytics_utility.md]: I think we can delete all the AVD creation. 
Android developers already have it :)

File: book/analytics/analytics_utility/analytics_utility.md [hunk 5]
```diff
@@ -21,9 +18,42 @@ Analytics data that is delivered to third party analytic providers are sent thro
...
+7. If you don’t see the logs, tap on the arrow here:
+	![logs_ios_7](./logs_ios_7.png)
+8. Start the app. Now the logs should reflect what is going on in the app.
+9. You can filter for analytics related activity.
+To do so, tap on the console panel and hit cmd+f for the search line. Enter there “Analytics Log:”
+10. Now, all activity related analytics handling will appear in the monitor, which you can use to assist with your debugging, QA, and development testing.
+
+## Android
 
 Download the app in debug mode. When you launch the app an alert will appear, asking if you want to receive notifications on analytics data delivered. If you hit yes, you'll be able to use the utility within the app directly.
```
Comment (snooky23) [inline book/analytics/analytics_utility/analytics_utility.md]: Can you put a screenshot for the alert?

File: book/analytics/analytics_utility/analytics_utility.md (hunk 4 above)
Comment (liviur) [reply book/analytics/analytics_utility/analytics_utility.md]: This is not necessarily aimed at developers.

Comment (liviur) [reply book/analytics/analytics_utility/analytics_utility.md]: at least not of the same platform

File: book/analytics/analytics_utility/analytics_utility.md (hunk 5 above)
Comment (liviur) [reply book/analytics/analytics_utility/analytics_utility.md]: it's android - if you can - please. not sure it's critical though

**Comment by liviur** [comment]
@YoniOsteen please continue to add comments etc. I'll open a separate PR to handle them

File: book/analytics/mobile_web_support/mobile_web_support.md [hunk 6]
```diff
@@ -66,6 +66,6 @@ To test your work on Android:
...
-    * Most analytics providers (Flurry, Google Analytics, Facebook Analytics, etc.) do not enable you to send PII (Personally Identifiable Information) or profile-level user data. Applicaster will still send events to these providers. However, for user profiles, Applicaster supports [Mixpanel](https://mixpanel.com/), an analytics tool which both supports the delivery of PII data and provides engagement and remarketing tools based on this data. If you want to test the delivery of user profiles, make sure to configure a test project for Mixpanel in Zapp. 
+    * Most analytics providers (Firebase Analytics, Facebook Analytics, etc.) do not enable you to send PII (Personally Identifiable Information) or profile-level user data. Applicaster will still send events to these providers. However, for user profiles, Applicaster supports [Mixpanel](https://mixpanel.com/), an analytics tool which both supports the delivery of PII data and provides engagement and remarketing tools based on this data. If you want to test the delivery of user profiles, make sure to configure a test project for Mixpanel in Zapp.
```
Comment (YoniOsteen) [inline book/analytics/mobile_web_support/mobile_web_support.md]: "If you want to test the delivery of user profiles which includes PII, make sure to configure a test project for Mixpanel or another provider who supports PII delivery in Zapp."

File: book/analytics/iOS.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+```
+##### User properties
+Set user profile properties.
+
+``` swift
+  @objc func updateGenericUserProperties(_ genericUserProfile: [String:NSObject])
+```
+
+##### In depth user profile properties
+An analytics provider can further set parameters for a specific user profile including optionally seperating PII properties by implementing the following optional functions:
```
[the remark this answers is not in the document]
Comment (liviur) [reply book/analytics/iOS.md]: 🤦‍♂

[the remark this answers is not in the document]
Comment (liviur) [reply book/analytics/iOS.md]: function exists on iOS... I assume they didn't just add a signature ...

**Pull request merged**